pkg/http/handler/timeout.go

```diff
@@ -31,6 +31,13 @@ import (
 // TimeoutFunc returns the timeout duration to be used by the timeout handler.
 type TimeoutFunc func(req *http.Request) time.Duration
 
+// StaticTimeoutFunc returns a TimeoutFunc that always returns the same duration.
+func StaticTimeoutFunc(timeout time.Duration) TimeoutFunc {
+	return func(req *http.Request) time.Duration {
+		return timeout
+	}
+}
+
```

Comment:
do we even need it here and public?

Reply:
Used in the QP.

Reply:
yeah I know. Why wouldn't it live right there?

Reply:
because I also used it in tests.

Reply:
meh.